Libplanet.Extensions.Cocona/BlockPolicyParams.cs

```diff
@@ -180,17 +186,79 @@ internal ImmutableArray<IAction> GetEndBlockActions(Assembly[] assemblies)
         if (propertyInfo is null)
         {
             var message = $"The policy type "
-                + $"'{policy.GetType().FullName}' does not have a "
-                + $"'{nameof(IBlockPolicy.EndBlockActions)}' property.";
+                          + $"'{policy.GetType().FullName}' does not have a "
+                          + $"'{nameof(IBlockPolicy.EndBlockActions)}' property.";
             throw new InvalidOperationException(message);
         }
 
         var value = propertyInfo.GetValue(policy);
         if (value is null)
         {
             var message = $"The value of property "
-                + $"'{nameof(IBlockPolicy.EndBlockActions)}' of type "
-                + $"'{policy.GetType().FullName}' cannot be null.";
+                          + $"'{nameof(IBlockPolicy.EndBlockActions)}' of type "
+                          + $"'{policy.GetType().FullName}' cannot be null.";
+            throw new InvalidOperationException(message);
+        }
+
+        return (ImmutableArray<IAction>)value;
+    }
+
+    internal ImmutableArray<IAction> GetBeginTxActions(Assembly[] assemblies)
+    {
+        object? policy = GetBlockPolicy(assemblies);
+        if (policy is null)
+        {
+            return ImmutableArray<IAction>.Empty;
+        }
+
+        PropertyInfo? propertyInfo = policy
+            .GetType()
+            .GetProperty(nameof(IBlockPolicy.BeginTxActions));
+        if (propertyInfo is null)
+        {
+            var message = $"The policy type "
+                          + $"'{policy.GetType().FullName}' does not have a "
+                          + $"'{nameof(IBlockPolicy.BeginTxActions)}' property.";
+            throw new InvalidOperationException(message);
+        }
+
+        var value = propertyInfo.GetValue(policy);
+        if (value is null)
+        {
+            var message = $"The value of property "
+                          + $"'{nameof(IBlockPolicy.BeginTxActions)}' of type "
+                          + $"'{policy.GetType().FullName}' cannot be null.";
+            throw new InvalidOperationException(message);
+        }
+
+        return (ImmutableArray<IAction>)value;
+    }
+
+    internal ImmutableArray<IAction> GetEndTxActions(Assembly[] assemblies)
```

Comment:
Don't we need any two or more types of policy actions? e.g., `GetBeginTxAndEndTxActions`.